### DISCUSSION TITLE
[V5] How to request local routing in the program?

### DISCUSSION THREAD

**Discussion by AGDholo** [adonisjs/core#1710] [Help] [answered]
Example:

```
Route.get('test', async () => {
  return test
})
```
```
Route.get('/', async () => {
  // Here or in controller I want to request the `/test`
})
```

In Laravel I can use the Request::create('test') to send the request

**Comment by RodolfoSilva** [comment, marked as answer]
Why not use a service or a command to do this?

**Comment by thetutlage** [comment]
As @RodolfoSilva said, using a Service to abstract the logic is a better option vs making an internal HTTP request

**Comment by AGDholo** [comment]
I am trying to use them, thank you!
@RodolfoSilva @thetutlage

**Reply by RodolfoSilva** [reply]
I thinks this article written by @shemsiu can help you more @AGDholo.

https://dev.to/shemsiu/ioc-container-and-dependency-injection-in-adonis-v5-og2

**Reply by AGDholo** [reply]
Very useful, I am trying to create it!
